.circleci/config.yml

```diff
@@ -1,28 +1,135 @@
-# Javascript Node CircleCI 2.0 configuration file
+# Javascript Node CircleCI 2.1 configuration file
 #
 # Check https://circleci.com/docs/2.0/language-javascript/ for more details
 #
-version: 2
+version: 2.1
+executors:
+  docker-publisher:
+    environment:
+      IMAGE_NAME: italia/developers-italia-onboarding
+    docker:
+      - image: circleci/buildpack-deps:stretch
 jobs:
   build:
     docker:
-      - image: circleci/node:8 
-      
-    working_directory: ~/repo
+      - image: circleci/node:8
 
+    working_directory: ~/repo
     steps:
       - checkout
-
-      # npm build 
-      - run: 
-          name: npm install 
+      - run:
+          name: npm install
           command: npm install
       - run:
-          name: npm lint 
+          name: npm lint
           command: npm run lint
+  docker-build-prod:
+    executor: docker-publisher
+    steps:
+      - checkout
+      - setup_remote_docker
+      - run:
+          name: Build Docker image
+          command: |
+            docker build -t $IMAGE_NAME:latest .
+      - run:
+          name: Archive Docker image
+          command: docker save -o image_prod.tar $IMAGE_NAME
+      - persist_to_workspace:
+          root: .
+          paths:
+            - ./image_prod.tar
+  docker-build-dev:
+    executor: docker-publisher
+    steps:
+      - checkout
+      - setup_remote_docker
+      - run:
+          name: Build Docker image
+          command: |
+            docker build -t $IMAGE_NAME:development .
+      - run:
+          name: Archive Docker image
+          command: docker save -o image_dev.tar $IMAGE_NAME
+      - persist_to_workspace:
+          root: .
+          paths:
+            - ./image_dev.tar
+  deploy-prod:
+    machine:
+      enabled: true
+    steps:
+      - attach_workspace:
+          at: /tmp/workspace
+      - run: scp /tmp/workspace/image_prod.tar circleci@$SSH_HOST:/tmp/image_prod.tar
```

Comment:
Since this is a tar file stored in /tmp, I would use a more appropriate name than `image_prod`, especially on the target machine. Something like `onboarding_docker_image_prod` that specifies what's inside that tar file and avoid useless confusion. 
Using a variable could be cool but I don't know if that's even possible.

Reply:
agree, and yes, we can use variable. [Here](https://circleci.com/docs/2.0/env-vars/) we have a list of env vars scope